### PR TITLE
Add saving led position images

### LINKED ISSUE #9: Save configuration reference images for better debugging
In WIP code we save just a reference image with all the lights turned on from every perspective, which has to be manually renamed before capturing the next side.

It would be great to have the option (controlled by a CLI flag possibly) to save an image of every single light separately, for possible review later. Currently we're left guessing why a mark might have been left in any given spot, since we don't see the actual image that was fed to OpenCV.

### PR DESCRIPTION
**Pull request by krzmaz**
Closes #9

### CODE REVIEW

File: configurator/src/capture.rs [hunk 1]
```diff
@@ -61,8 +61,17 @@ impl Capturer {
...
     pub async fn capture_perspective(
         &mut self,
+        save_pictures: bool,
     ) -> Result<Vec<Option<(usize, usize)>>, Box<dyn Error>> {
         let mut coords = Vec::new();
+        let timestamp = SystemTime::now()
+            .duration_since(SystemTime::UNIX_EPOCH)?
+            .as_millis()
+            .to_string();
+        let dir = format!("captures/{}", timestamp.as_str());
```
Comment (mrozycki) [inline configurator/src/capture.rs:71]: nit: timestamp is already a string, I don't thing you need the as_str

File: configurator/src/capture.rs [hunk 2]
```diff
@@ -61,8 +61,17 @@ impl Capturer {
 
     pub async fn capture_perspective(
         &mut self,
+        save_pictures: bool,
     ) -> Result<Vec<Option<(usize, usize)>>, Box<dyn Error>> {
         let mut coords = Vec::new();
+        let timestamp = SystemTime::now()
+            .duration_since(SystemTime::UNIX_EPOCH)?
+            .as_millis()
```
Comment (mrozycki) [inline configurator/src/capture.rs:69]: Wouldn't it be more readable in the ISO format? Year, month, day, hour, minutes, seconds. Also not sure if we need it down to millis, it's not like you'll be able to capture more than 1 in one second.

File: configurator/src/capture.rs [hunk 3]
```diff
@@ -94,9 +103,21 @@ impl Capturer {
...
             thread::sleep(Duration::from_millis(30));
-            let led_picture = self.camera.capture()?;
-
-            coords.push(cv::find_light_from_diff(&base_picture, &led_picture)?);
+            let mut led_picture = self.camera.capture()?;
+            let found_coords = cv::find_light_from_diff(&base_picture, &led_picture)?;
+            if save_pictures {
+                if found_coords.is_some() {
+                    led_picture.mark(found_coords.unwrap().0, found_coords.unwrap().1)?;
+                }
```
Comment (mrozycki) [inline configurator/src/capture.rs:111]:
```suggestion
                if let Some((x, y)) = found_coords {
                    led_picture.mark(x, y)?;
                }
```

File: configurator/src/capture.rs [hunk 4]
```diff
@@ -94,9 +103,21 @@ impl Capturer {
...
+                let filename = format!("{}/{}.jpg", dir, i);
```
Comment (mrozycki) [inline configurator/src/capture.rs:112]:
```suggestion
                let filename = format!("{}/{:03}.jpg", dir, i);
```

This way they'll be padded with zeroes to 3 digits, and so will end up in alphabetical order. Easier to browse.

File: configurator/src/cv.rs [hunk 5]
```diff
@@ -28,7 +26,7 @@ impl Error for CameraError {}
 
 #[derive(Default)]
 pub struct Picture {
-    inner: Mat,
+    pub inner: Mat,
 }
```
Comment (mrozycki) [inline configurator/src/cv.rs:30]: I feel like maybe a better solution would be to make a `save_file` method on the `Picture` and use that in the other file. Less code duplication as well.

File: configurator/src/capture.rs (hunk 2 above)
Comment (krzmaz) [reply configurator/src/capture.rs]: Done

File: configurator/src/capture.rs (hunk 3 above)
Comment (krzmaz) [reply configurator/src/capture.rs]: Thanks, I was pretty sure there was a better option for doing that 😁

File: configurator/src/capture.rs (hunk 4 above)
Comment (krzmaz) [reply configurator/src/capture.rs]: Done

File: configurator/src/cv.rs (hunk 5 above)
Comment (krzmaz) [reply configurator/src/cv.rs]: Done

**Review by mrozycki — APPROVED** [review]

**Pull request merged**